### ISSUE TITLE
Tweets Don't Lead to Right Page

### ISSUE THREAD

**Issue by kolatubosun** [Yorubaname/yorubaname-website#32]
It might be a result of the tone marking or any other bug. But when users tweet from within the entry, the tweeted link doesn't lead back to the entry. Check out the screenshots.

![screenshot 25](https://cloud.githubusercontent.com/assets/18510601/16726680/6b1248d8-4755-11e6-9983-f21944a085d3.png)
![screenshot 24](https://cloud.githubusercontent.com/assets/18510601/16726686/724e9318-4755-11e6-83f0-dac6668a4727.png)

**Comment by kolatubosun** [comment]
I realise now that the problem might also be because the web link is cut off in the tweet, so when one clicks, it only seeks the part that shows, instead of the full link. If this is the problem, a solution might be to have the tweet pass through a link shortener (like bit.ly or goo.gl) so that the tweet can fit the full link.

I don't know if this is the case or the solution, but it's worth examining.

**Comment by zeddysoft** [comment]
@kolatubosun hope no one is working on this feature yet, this looks like something i want to work on

**Comment by kolatubosun** [comment]
Hi @zeddysoft Please go ahead an work on it, but let's ask @dadepo or @temiyemi if the issue is linked to any other internal issue anyone else is working on.

**Comment by dadepo** [comment]
Hi @zeddysoft I am not working on this issue. And as far as I can tell, it is not linked to any other issue. So feel free to pick this up :)

**Comment by temiyemi** [comment]
No, this is not currently linked to any of the ongoing issues being fixed. But this behaviour would change once we make the changes discussed last weekend: re: not using tonal marks in the URLs etc.

**Comment by kolatubosun** [comment]
" this behaviour would change once we make the changes discussed last weekend: re: not using tonal marks in the URLs etc."

That's what I was wondering about.

**Comment by dadepo** [comment]
@temiyemi correct me if I am wrong, the issue you refer to is issue #42 ? If so, I guess @zeddysoft  you can read up on the said conversation. it might shape how you approach this issue then. I myself have not had time to catch up on the discussions in that issue, once I do, I will share my perspective.

**Comment by temiyemi** [comment]
@dadepo you're correct. Fixing #42 would sort this out going forward. But old links (with tonal marks) may not resolve after the change.

**Comment by dadepo** [comment]
@temiyemi based on the comment I just left on issue #42 (https://github.com/Yorubaname/yorubaname-website/issues/42#issuecomment-242846465), most likely, this would still be a valid issue that needs fixing.

**Comment by temiyemi** [comment]
I get your drift. I kind of agree. The original issue on #42 is updating the entries. Your suggestion should fix that. But we also need to implement such that tonal marks are not the primary identifier for names spelt the same way. And when we do that, it would actually fix this URL issue.

**Comment by dadepo** [comment]
> "But we also need to implement such that tonal marks are not the primary identifier for names spelt the same way" 

True. That might end up being a thing we do. But when I briefly looked into this issue when it was first created, my conclusion was that, it has nothing to do with the application per say but with the twitter snippets that allows names to be shared (not configured correctly to support characters with accents etc) because if you copy the link to the name and paste it directly in a tweet yourself, it does actually link back to the entry in the dictionary when you click it.

Even if it ends up being an issue with the application itself it looks like the fix would most likely be something like setting the correct character encoding. 

It thus feel like changing the behaviour regarding the role tonal marks play as regards identifying an entry is an overkill - unless though, there are other benefits (ux improvement etc) if we make that change. But it would be for those benefits then, and not because we want to fix this particular issue.

**Comment by temiyemi** [comment]
@dadepo how might we integrate https://github.com/assemble/handlebars-helpers so we can use some of the handlebars helpers? The URLEncode helper might help fix this one

**Comment by Flipponachi** [comment]
@dadepo I tried copying and pasting into my browser and it doesn't lead back to the entry. My best guess is that its from Twitter's API itself, but the funny thing in the first instance is that it displays correctly on twitter timeline and yet it doesn't direct back. I've opened issues with the Twitter dev community and I should get a response soon, the other option we have is to create a parse function itself and put it onto the website as a middleware and then it corrects the format. Its a long shot and am open to other ideas, I really love this bug already!!
